### PR TITLE
Added Header Files and Rearranged Folder Structure for Vision

### PR DESCRIPTION
**Pull request by priankakariat**
Added header files
1. ios/task/processor/TFLClassificationResult.h
2. ios/TFLCommonUtil.h

Rearranged public headers into apis and internal headers into sources for vision

### CODE REVIEW

File: tensorflow_lite_support/ios/sources/TFLCommonUtils.h [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#import <Foundation/Foundation.h>
+#import <TensorFlowLiteTaskVisionC/TensorFlowLiteTaskVisionC.h>
```
Comment (lu-wang-g) [inline tensorflow_lite_support/ios/sources/TFLCommonUtils.h:17]: Is this needed?

File: tensorflow_lite_support/ios/sources/TFLCommonUtils.h [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ *
+ * @param memSize size of memory to be allocated
+ * @param error Pointer to the memory location where errors if any should be saved. If `nil`, no
+ * error will be saved.
+ *
+ * @return Pointer to the allocated block of memory on successfull allocation. nil in case as error
+ * is encountered because of invalid memSize. If failure is due to any other reason, method
+ * terminates program execution.
+ */
++ (void *)mallocOrErrorWithSize:(size_t)memSize error:(NSError **)error;
```
Comment (lu-wang-g) [inline tensorflow_lite_support/ios/sources/TFLCommonUtils.h:60]: Should this be  `mallocOrErrorWithSize` or `mallocErrorWithSize`?

File: tensorflow_lite_support/ios/task/processor/apis/TFLClassificationOptions.h [hunk 3]
```diff
@@ -12,7 +12,7 @@
  See the License for the specific language governing permissions and
  limitations under the License.
  ==============================================================================*/
-
+
```
Comment (lu-wang-g) [inline tensorflow_lite_support/ios/task/processor/apis/TFLClassificationOptions.h:15]: nit: remove the empty space.

File: tensorflow_lite_support/ios/task/processor/apis/TFLClassificationResult.h [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/** Encapsulates information about a class in the classification results. */
+@interface TFLCategory : NSObject
+
+/** Display name of the class. */
+@property(nonatomic, strong) NSString *displayName;
```
Comment (khanhlvg) [inline tensorflow_lite_support/ios/task/processor/apis/TFLClassificationResult.h:24]: As this is an entity class, I think we should use `copy` is better than `strong` to make sure that the values aren't changed unintentionally. Same for other properties of object type.

File: tensorflow_lite_support/ios/sources/TFLCommonUtils.h (hunk 1 above)
Comment (khanhlvg) [reply tensorflow_lite_support/ios/sources/TFLCommonUtils.h:16]: Thanks for the clarification. SGTM.

Comment (lu-wang-g) [reply tensorflow_lite_support/ios/sources/TFLCommonUtils.h:16]: What is "TensorFlowLiteTaskVisionC.h"? It's a little wired to have TFLCommonUtils.h depends on Task Vision C API.

**Pull request merged**